src/payments/TransferHelper.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import {SafeERC20} from "openzeppelin/token/ERC20/utils/SafeERC20.sol";
+import {IWETH, IERC20} from "./IWETH.sol";
+
+/*//////////////////////////////////////////////////////////////////////////
+                            Custom Errors
+//////////////////////////////////////////////////////////////////////////*/
+
+/// @dev ETH transfer failed
+error ETHTransferFailed();
+
+/// @dev Transferred too few ERC-20 tokens
+error InsufficentERC20Transfer();
+
+/*//////////////////////////////////////////////////////////////////////////
+                            Transfer Helper
+//////////////////////////////////////////////////////////////////////////*/
+
+/// @title Transfer Helper
+/// @notice Abstract contract that has helper function for sending ETH and ERC20's safely
+/// @author transientlabs.xyz
+/// @custom:last-updated 2.3.0
+abstract contract TransferHelper {
+    /*//////////////////////////////////////////////////////////////////////////
+                                  State Variables
+    //////////////////////////////////////////////////////////////////////////*/
+
+    using SafeERC20 for IERC20;
+    using SafeERC20 for IWETH;
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                   ETH Functions
+    //////////////////////////////////////////////////////////////////////////*/
+
+    /// @notice Function to force transfer ETH
+    /// @dev On failure to send the ETH, the ETH is converted to WETH and sent
+    /// @dev Care should be taken to always pass the proper WETH address that adheres to IWETH
+    /// @param recipient The recipient of the ETH
+    /// @param amount The amount of ETH to send
+    /// @param weth The WETH token address
+    function _safeTransferETH(address recipient, uint256 amount, address weth) internal {
```

Comment:
love the idea, one thing to consider is if a dev uses a custom contract that doesnt have a concept of weth and that value gets locked up. although with an escrow this may be the outcome anyway